### ISSUE TITLE
Webpack 4: Cannot read property 'resolveUrlLoader' of undefined

### ISSUE THREAD

**Issue by acoll** [bholloway/resolve-url-loader#79]
It looks like loader.options is undefined on https://github.com/bholloway/resolve-url-loader/blob/master/index.js#L39 when using webpack 4.0.0-beta.2.

I'm not sure if this is a problem with resolver-url-loader or with webpack.

**Comment by ranyitz** [comment]
@acoll This is something that was removed in webpack 4

> `this.options` and `this.rootContext`
webpack 3 already deprecated `this.options` in the loader context. webpack 4 removes it now.

> Guideline: Loaders should receive all options via `this.query`. They should not use other ways to receive options, i. e. no property in webpack options, no environment variable.

@bholloway This is breaking for people that try to migrate to webpack 4.

+1 for #80

**Comment by acoll** [comment]
Ok cool. Could you link to where you pulled that guideline? So based on that should the options be computed as `defaults(loader.query, {/* ... the defaults */})`?

I tried that locally and it seems to make more sense than my other change and loader.query contains the options i'm setting in my webpack config.

**Comment by ranyitz** [comment]
Unfortunately, there are no specific anchors, but it's in the [release notes of webpack 4](https://github.com/webpack/webpack/releases/tag/v4.0.0-beta.0).

About your comment, `loaderUtils` actually [does that](https://github.com/webpack/loader-utils#getoptions).

**Comment by petrleocompel** [comment]
Now when webpack [4.0.0](https://github.com/webpack/webpack/releases/tag/v4.0.0) was released. 
See also `Breaking changes for plugins/loaders` on release page.

**Comment by bholloway** [comment]
Thanks everyone.

Unfortunately I’m AFK for another week. Sorry I didn’t anticipate this.

Let me see what I can do to get the fix published in the meantime.

**Comment by julmot** [comment]
I also have this issue 

PR: https://github.com/julmot/form-components/pull/9
Build: https://travis-ci.org/julmot/form-components/builds/347651941?utm_source=github_status&utm_medium=notification

**Comment by julmot** [comment]
@bholloway What's the status?

**Comment by hadnet** [comment]
Same thing here.

**Comment by dmaicher** [comment]
I'm using this fix temporarily: 

https://github.com/dmaicher/resolve-url-loader/commit/1e9d6f046e7efb80988a3f8b4da89067036d6127

```
"resolve-url-loader": "github:dmaicher/resolve-url-loader#1e9d6f046e7efb80988a3f8b4da89067036d6127"

**Comment by bholloway** [comment]
Sorry for the delay. I'm hot off the plane and at the keyboard right now. Please follow #80.

**Comment by bholloway** [comment]
Thanks to @acoll the fix is now on `master`. If anyone is available please have a **quick test of `master` before I publish**.

There are only minor changes to what was tested in #80 and _should_ just work. I have checked a legacy Webpack 1 project and that is good but that is all I have to hand right now.

(My apologies I really need to get a good matrix of automated testing)

**Comment by alexdevero** [comment]
Thanks a lot @bholloway! I just tested it on one of my apps and the unpublished `bholloway/resolve-url-loader` works.

**Comment by cascornelissen** [comment]
The latest `master` version seems to work on my end as well (webpack 4) 👍

**Comment by bholloway** [comment]
Thanks @alexdevero @cascornelissen. Published as `2.3.0`.

Please let me know any immediate problems here. Thanks for the contribution to this issue.

**Comment by julmot** [comment]
@bholloway The release is not on GitHub, only on npm.
